utils/githooks/pre-commit.d/40-gofmt.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+# Runs pylint for the DAOS project as a commit hook.
+#
+# To get the most out of this hook the 'gh' command should be installed and working.
+
+set -ue
+
+if ! BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null); then
+        echo "Failed to determine branch with git rev-parse"
+        exit 1
+fi
+
+# Try and use the gh command to work out the target branch, or if not installed
+# then assume master.
+if command -v gh > /dev/null 2>&1
+then
+        # If there is no PR created yet then do not check anything.
+        if ! TARGET=origin/$(gh pr view "$BRANCH" --json baseRefName -t "{{.baseRefName}}")
+        then
+                TARGET=HEAD
+        fi
+else
+        # With no 'gh' command installed then check against master.
+        echo "Install gh command to auto-detect target branch, assuming master."
+        TARGET=origin/master
+fi
+
+output=
+if [ $TARGET = "HEAD" ]
+then
+        echo "Checking against HEAD"
+        output=$(git diff HEAD^ --name-only | grep -e '.go$' | xargs gofmt -d)
+else
+        echo "Checking against branch ${TARGET}"
+        output=$(git diff $TARGET... --name-only | grep -e '.go$' | xargs gofmt -d)
+fi
```

Comment:
Not saying necessarily for this PR, but this code is begging to be DRYed out.  We have it in a number of hooks now.